addons/mail/static/tests/discuss_app/sidebar.test.js

```diff
@@ -1174,15 +1165,9 @@ test("Update channel data via bus notification", async () => {
 test("sidebar: show loading on initial opening", async () => {
     // This could load a lot of data (all pinned conversations)
     const def = new Deferred();
-    onRpcBefore("/mail/action", async (args) => {
-        if (args.fetch_params.includes("channels_as_member")) {
-            await def;
-        }
-    });
-    onRpcBefore("/mail/data", async (args) => {
-        if (args.fetch_params.includes("channels_as_member")) {
-            await def;
-        }
+    const { waitStoreFetch } = listenStoreFetch("channels_as_member", async () => {
+        asyncStep("before channels_as_member");
```

Comment:
Note: this step is necessary to ensure the callback is actually applied, it makes the test much more robust.